lib/krane/kubernetes_resource/deployment.rb

```diff
@@ -191,7 +191,7 @@ def find_latest_rs(cache)
     def min_available_replicas
       if percent?(required_rollout)
         (desired_replicas * required_rollout.to_i / 100.0).ceil
-      elsif max_unavailable =~ /%/
+      elsif max_unavailable.is_a?(String) && max_unavailable =~ /%/
```

Comment:
Why the `.is_a?(String)` ?

Reply:
To get rid of this warning:
```
[1] pry(main)> obj = 5; obj =~ /%/
(pry):1: warning: deprecated Object#=~ is called on Integer; it always returns nil
```
It shows up in the test suite, and it seems likely it can be either an integer or string (e.g. `"25%"`) in production.